simulator/mips/t/unit_test.cpp

```diff
@@ -236,10 +146,2252 @@ TEST_CASE( "MIPS32_instr_disasm: Process_Disasm_Branches")
+    instr.set_v_src( 1, 0);
+    instr.set_v_src( 64, 1);
+    instr.execute();
+    CHECK( instr.get_v_dst() == 1);
+}
+// ***********************************/
+////////////////////////////////////////////////////////////////////////////////
+
+TEST_CASE( "MIPS32_instr: j test_1")
+{
+    CHECK(MIPS32Instr(0x0bfffb2e).get_disasm() == "j 0x3fffb2e");
+    CHECK(MIPS32Instr(0x080004d2).get_disasm() == "j 0x4d2");
+    
+    MIPS32Instr instr( "j");
+    instr.set_v_imm( 4);
+    instr.execute();
+    CHECK( instr.get_new_PC() == 16);
+}
+////////////////////////////////////////////////////////////////////////////////
+
+TEST_CASE( "MIPS32_instr: jal test_1")
+{
+    CHECK(MIPS32Instr(0x0ffffb2e).get_disasm() == "jal 0x3fffb2e");
+    CHECK(MIPS32Instr(0x0c0004d2).get_disasm() == "jal 0x4d2");
+    
+    MIPS32Instr instr( "jal");
+    instr.set_v_imm( 0xfff);
+    instr.execute();
+    CHECK( instr.get_new_PC() == 0xfff * 4);
+    CHECK( instr.get_v_dst() == instr.get_PC() + 4);
+}
+////////////////////////////////////////////////////////////////////////////////
+
+TEST_CASE( "MIPS32_instr: jalr test_1")
+{
+    CHECK(MIPS32Instr(0x01208809).get_disasm() == "jalr $s1, $t1");
+
+    MIPS32Instr instr( "jalr");
+    instr.set_v_src( 1024, 0);
+    instr.execute();
+    CHECK( instr.get_new_PC() == 1024);
+    CHECK( instr.get_v_dst() == instr.get_PC() + 4);
+}
+
+TEST_CASE( "MIPS32_instr: jalr test_2")
+{
+    MIPS32Instr instr( "jalr");
+    instr.set_v_src( 2, 0);
+    instr.execute();
+    CHECK( instr.get_new_PC() == 4);
+    CHECK( instr.get_v_dst() == instr.get_PC() + 4);
+}
+////////////////////////////////////////////////////////////////////////////////
+
+TEST_CASE( "MIPS32_instr: jr test_1")
+{
+    CHECK(MIPS32Instr(0x02200008).get_disasm() == "jr $s1");
+    
+    MIPS32Instr instr( "jr");
+    instr.set_v_src( 0xfffffff4, 0);
+    instr.execute();
+    CHECK( instr.get_new_PC() == 0xfffffff4);
+}
+
+TEST_CASE( "MIPS32_instr: jr test_2")
+{
+    MIPS32Instr instr( "jr");
+    instr.set_v_src( 2, 0);
+    instr.execute();
+    CHECK( instr.get_new_PC() == 4);
+}
+////////////////////////////////////////////////////////////////////////////////
+
+TEST_CASE( "MIPS32_instr: mfhi test_1")
+{
+    CHECK(MIPS32Instr(0x00008810).get_disasm() == "mfhi $s1");
+    
+    MIPS32Instr instr( "mfhi");
+    instr.set_v_dst( 0);
+    instr.set_v_src( 0xdeadbeef, 0);
+    instr.execute();
+    CHECK( instr.get_v_dst() == 0xdeadbeef);
+}
+////////////////////////////////////////////////////////////////////////////////
+
+TEST_CASE( "MIPS32_instr: mflo test_1")
+{
+    CHECK(MIPS32Instr(0x00008812).get_disasm() == "mflo $s1");
+    
+    MIPS32Instr instr( "mflo");
+    instr.set_v_dst( 0);
+    instr.set_v_src( 0xfee1dead, 0);
+    instr.execute();
+    CHECK( instr.get_v_dst() == 0xfee1dead);
+}
+////////////////////////////////////////////////////////////////////////////////
+
+/*      Not implemented on the register file level
```

Comment:
Those may be deleted. We test masked writes on RF level instead.

Reply:
Done.